### PR TITLE
#839 Radvance: Check user exists on /permission add

### PR DESCRIPTION
**Pull request by prajapati-kaushik**

### CODE REVIEW

File: src/Controller/PermissionController.php [hunk 1]
```diff
@@ -34,18 +34,33 @@ public function addAction(Application $app, Request $request, EventDispatcherInt
         $space = $app->getSpaceRepository()->findByNameAndAccountName($spaceName, $accountName);
 
         $error = null;
-        if ($space) {
-            $repo = $app->getPermissionRepository();
-            $error = $repo->add($username, $space->getId(), $roles);
-            $admin = $app['current_user']->getName();
-            $event = new PermissionDomain\PermissionGrantedEvent(
+
+        $client = $app['userbase.client'];
```
Comment (h-wang) [inline src/Controller/PermissionController.php]: @kaushikindianic the `indexAction` should also be protected, not only the `addAction`

File: src/Controller/PermissionController.php [hunk 2]
```diff
@@ -34,18 +34,33 @@ public function addAction(Application $app, Request $request, EventDispatcherInt
...
+        try {
+            $user = $client->getUserByUsername($username);
```
Comment (h-wang) [inline src/Controller/PermissionController.php]: @kaushikindianic `$username` is the posted value. we should check `$app['current_user']`. 
For now, as long as the user is logged in, we can allow the index and add actions.

**Comment by h-wang** [comment]
@joostfaassen the validation of `user` should refer to the `userSecurityProvider`. In other words, in this context, `user` means the accounts loadable by the `userSecurityProvider`. 

The `Symfony\Component\Security\Core\User\UserProviderInterface` provides only `loadUserByUsername($username)`. 
So I think instead of `$userBaseClient->getUserByUsername($username)`, it should invoke `$app->getUserSecurityProvider()->loadUserByUsername($username)`

**Comment by joostfaassen** [comment]
@h-wang @kaushikindianic agreed /w Hongliang's suggested solution

**Pull request merged**